include/sys/spa_impl.h

```diff
@@ -270,6 +270,7 @@ struct spa {
 	uint64_t	spa_scan_pass_scrub_spent_paused; /* total paused */
 	uint64_t	spa_scan_pass_exam;	/* examined bytes per pass */
 	uint64_t	spa_scan_pass_issued;	/* issued bytes per pass */
+	boolean_t	spa_resilver_defered;	/* resilver has been defered */
```

Comment:
typo: "defered" should be "deferred"

Reply:
This idea has merit.

But, the other side is if you are re-silvering a 12TB disk that is 9TBs full, you may not want to re-start at any point. Except perhaps very early on. This only gets worse as larger drives are released.

Two ways to handle this:
1) Time based. If the existing is going to finish soonish, (a debatable time...), then continue.
2) Size / percent based.

Another way to handle this is for the feature flag to have a companion variable that specifies percent complete for restarting. Perhaps defaulting to 0 / zero for deferring all re-silvers. And 100 for allowing a second to cause a restart at any point.